### PR TITLE
Make changes for GHC 9.6

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: src/Fusion/Plugin.hs [hunk 1]
```diff
@@ -712,8 +724,11 @@ fusionSimplify _hsc_env dflags =
             , sm_float_enable = floatEnable dflags
 #endif
             }
+    -}
     in CoreDoSimplify
-#if MIN_VERSION_ghc(9,5,0)
+#if MIN_VERSION_ghc(9,6,1)
+        (initSimplifyOpts dflags [] (maxSimplIterations dflags) mode undefined)
```
Comment (harendra-kumar) [inline src/Fusion/Plugin.hs]: TODO

**Comment by harendra-kumar** [comment]
@pranaysashank can you take a look at this, do you have an idea how to fix this.

Also, using your approach to reduce CPP, we can perhaps keep the current implementation in a single file for all versions before 9.6 and start creating a separate file from 9.6 onwards. It may not be a good idea to split the older versions, as those will not see any change in future.

Also, see https://github.com/composewell/fusion-plugin/issues/52 , if we start removing code for older release, maybe we can bear with the CPP?

**Comment by pranaysashank** [comment]
> Also, using your approach to reduce CPP, we can perhaps keep the current implementation in a single file for all versions before 9.6 and start creating a separate file from 9.6 onwards.

That's a good idea, I was thinking the same

**Comment by pranaysashank** [comment]
> Also, see https://github.com/composewell/fusion-plugin/issues/52 , if we start removing code for older release, maybe we can bear with the CPP?

Removing code is also easier with a separate file, we just delete the file making reviews also easy

**Comment by harendra-kumar** [comment]
@adamconnersax please take a look and try it out.

**Pull request merged**